mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -431,6 +431,14 @@ class Word2Vec extends Serializable with Logging {
 class Word2VecModel private[mllib] (
     private val model: Map[String, Array[Float]]) extends Serializable with Saveable {
```

Comment:
Don't store this any more.  Store wordVecMat, plus a matching collection of words; that collection should probably be a Map[String, Int] mapping word to index in wordVecMat (so that transform() is still fast).

Naming: How about "wordVectors" instead of "wordVecMat"?

You'll need to update getVectors to construct the map.